src/Neo.VM/ScriptBuilder.cs

```diff
@@ -93,7 +93,8 @@ public ScriptBuilder EmitJump(OpCode opcode, int offset)
         /// <returns>A reference to this instance after the emit operation has completed.</returns>
         public ScriptBuilder EmitPush(BigInteger value)
         {
-            if (value >= -1 && value <= 16) return Emit(OpCode.PUSH0 + (byte)(int)value);
+            if (value >= 0 && value <= 16) return Emit(OpCode.PUSH0 + (byte)(int)value);
+            if (value == -1) return Emit(OpCode.PUSHM1);
```

Comment:
But there's no bug here, `-1` was handled properly with the old code because `(byte)(-1)` is `0xFF`, and `OpCode.PUSH0 + (byte)(int)value` is `0x10 + 0xFF` which gives us `0x0F` due to `byte` size constraints.

Reply:
Anna is right, the -1 opcode is properly defined as 0x0f to make it suit for this convert.......

Reply:
Yes, she is right, but I prefer to add the condition instead of the overflow use